modules/installation-special-config-kmod.adoc

```diff
@@ -0,0 +1,304 @@
+KMOD_NAMES="simple-kmod simple-procfs-kmod"
+----
+
+. Create an instance of kmods-via-containers@.service for your kernel module
+(simple-kmod in this example) and enable it:
++
+----
+$ sudo make install
+$ sudo kmods-via-containers build simple-kmod $(uname -r)
+----
+. Enable and start the systemd service, then check the status:
++
+----
+$ sudo systemctl enable kmods-via-containers@simple-kmod.service
+$ sudo systemctl start kmods-via-containers@simple-kmod.service
+$ sudo systemctl status kmods-via-containers@simple-kmod.service
+● kmods-via-containers@simple-kmod.service - Kmods Via Containers - simple-kmod
+   Loaded: loaded (/etc/systemd/system/kmods-via-containers@.service;
+          enabled; vendor preset: disabled)
+   Active: active (exited) since Sun 2020-01-12 23:49:49 EST; 5s ago...
+----
+
+. Here are a few ways to interact with the kernel modules in this example and make sure they were loaded:
++
+----
+$ lsmod | grep simple_
+simple_procfs_kmod     16384  0
+simple_kmod            16384  0
+$ dmesg | grep 'Hello world'
+[ 6420.761332] Hello world from simple_kmod.
+$ sudo cat /proc/simple-procfs-kmod
+simple-procfs-kmod number = 0
+$ spkut 44
+KVC: wrapper simple-kmod for 4.18.0-147.3.1.el8_1.x86_64
+Running userspace wrapper using the kernel module container...
++ podman run -i --rm --privileged
+   simple-kmod-dd1a7d4:4.18.0-147.3.1.el8_1.x86_64 spkut 44
+simple-procfs-kmod number = 0
+simple-procfs-kmod number = 44
+----
+
+Going forward, when the system boots this service will check if a new
+kernel is running. If there is a new kernel, the service builds a new
+version of the kernel module and then loads it. If the module is already
+built, it will just load it.
+
+== Provision the kernel module via Ignition
+Depending on whether or not you need to have the kernel module in place
+when OpenShift Container Platform cluster first boots, you can set up the
+kernel modules to be deployed in one of two ways:
+
+* **Provision kernel modules at cluster install time (day-1)**: You can add
+the kernel module software to Ignition config files during installation by
+including the required content through the `openshift-install` command.
+
+* **Provision kernel modules via MCO (day-2)**: If you can wait until the
+cluster is up and running to add your kernel module, you can the kernel
+module software via the Machine Config Operator (MCO).
+
+In either case, each node needs access to new kernel packages and related
+software packages as they are available. There are a few ways to provide this information:
+
+* Provide RHEL entitlements to each node (through `subscription-manager` or secrets)
+* Get RHEL entitlements from an existing RHEL host (`/etc/pki/entitlement` directory)
+and copy them to the same location in `$FAKEROOT` when you build your Ignition config
+* Inside the Dockerfile, add pointers to a yum repository containing the kernel and other packages
+(this must include new kernel packages as they are available)
```

Comment:
other packages. This must include new kernel packages as they are available.

Reply:
Done